pkg/apis/provisioning/v1alpha2/provisioner_validation.go

```diff
@@ -120,6 +122,31 @@ func (c *Constraints) validateLabels() (errs *apis.FieldError) {
 	return errs
 }
 
+func (c *Constraints) validateTaints() (errs *apis.FieldError) {
+	for i, taint := range c.Taints {
+		// Validate Key
+		if len(taint.Key) == 0 {
+			errs = errs.Also(apis.ErrInvalidArrayValue(errs, "taints", i))
+		}
+		for _, err := range validation.IsQualifiedName(taint.Key) {
+			errs = errs.Also(apis.ErrInvalidArrayValue(err, "taints", i))
+		}
+		// Validate Value
+		if len(taint.Value) != 0 {
+			for _, err := range validation.IsQualifiedName(taint.Value) {
+				errs = errs.Also(apis.ErrInvalidArrayValue(err, "taints", i))
+			}
+		}
+		// Validate effect
+		switch taint.Effect {
+		case v1.TaintEffectNoSchedule, v1.TaintEffectPreferNoSchedule, v1.TaintEffectNoExecute, "":
```

Comment:
I'm guessing not, but is there any way to avoid having to keep this up-to-date (as more `Effect`s are added)?

Reply:
Not in go. Other languages can do static checking on enum values, but go does not support this.